javascript/javascript1/week4/classExercises/sentimentAnalyzer.js

```diff
@@ -0,0 +1,47 @@
+const positive = [
+  "joyful",
+  "excited",
+  "serene",
+  "grateful",
+  "vibrant",
+  "optimistic",
+  "radiant",
+  "enthusiastic",
+  "blissful",
+  "peaceful",
+  "super",
+  "awesome",
+  "happy",
+];
+const negative = [
+  "gloomy",
+  "anxious",
+  "melancholic",
+  "frustrated",
+  "dreary",
+  "miserable",
+  "disheartened",
+  "despondent",
+  "agitated",
+  "pessimistic",
+];
+
+function getSentimentScore(string) {
+  const strArr = string.split(" ");
+  let obj = { score: 0, positiveWords: [], negativeWords: [] };
+  for (let str of strArr) {
+    if (positive.includes(str)) {
+      obj.positiveWords.push(str);
+      obj.score++;
+    } else if (negative.includes(str)) {
+      obj.negativeWords.push(str);
+      obj.score--;
+    }
+  }
+  return obj;
+}
+
+// Or score = obj.positiveWords.length - obj.negativeWords.length
+
+const sentimentScoreObject = getSentimentScore("I am mega super awesome happy");
```

Comment:
Grammatically those would be comma separated, but then `split(" ")` won't work. Another approach would be to skip making an array out of the initial string, and instead loop through the list of the positive and negative words and search for a match. Something along those lines:
```
  for (let i = 0; i < listPositiveWords.length; i++) {
    if (sentence.match(listPositiveWords[i])) {
      positiveScoreWords.push(listPositiveWords[i]);
      score++;
    }
  }
```